bootstrap/main.go

```diff
@@ -0,0 +1,209 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"regexp"
+	"strings"
+
+	flag "github.com/spf13/pflag"
+)
+
+func printUsage(flags *flag.FlagSet) {
+	log.Println("Flags")
+	log.Println(flags.FlagUsages())
+}
+
+// ParseArgs parses the command line to supported arguments as a pflag.FlagSet
+// arguments should be generally be os.Args[1:]
+func ParseArgs(arguments []string) *flag.FlagSet {
+	args := flag.NewFlagSet("bootstrap", flag.ExitOnError)
+	// used to mimic required=true
+	requiredFlags := []string{"job"}
+
+	// add all of the args from parse_args in jenkins/bootstrap.py
+	args.String("root", ".", "Root dir to work with")
+
+	// NOTE: jenkins/bootstrap.py technically used a float for this arg
+	// when parsing but only ever used the arg as an integer number
+	// of timeout minutes.
+	// int32 makes more sense and will let use time.Minute * timeout
+	args.Int32("timeout", 0, "Timeout in minutes if set")
+
+	// TODO(bentheelder): remove these?
+	args.String("pull", "", "Deprecated, use --repo=k8s.io/foo=master:abcd,12:ef12,45:ff65")
+	args.String("branch", "", "Deprecated, use --repo=k8s.io/foo=master")
+
+	args.StringArray("repo", []string{},
+		"Fetch the specified repositories, with the first one considered primary")
+
+	args.Bool("bare", false, "Do not check out a repository")
+	args.Lookup("bare").NoOptDefVal = "true" // allows using --bare
```

Comment:
I don't think this is necessary

Reply:
I wanted to mimic the behavior of the existing args which allows `--bare` or `--bare=true`, we could remove it but since it only took one line to support it didn't seem worth bikeshedding yet.

Reply:
Go does this by default